services/secondary-storage/server/src/main/java/org/apache/cloudstack/storage/resource/NfsSecondaryStorageResource.java

```diff
@@ -772,7 +780,12 @@ protected Answer copyFromSwiftToNfs(CopyCommand cmd, DataTO srcData, SwiftTO swi
                 }
             }
 
-            File destFile = SwiftUtil.getObject(swiftTO, downloadDirectory, srcData.getPath());
+            String filePath = downloadPath + File.separator + destData.getName();
```

Comment:
Are you not using the `getPath` to find the path of the object anymore? does `downloadPath` consider the difference between snapshot and template and now volume?

Reply:
Hey Syed, thanks for the review. We do call getPath() on line 766 so the downloadPath will contain the path to folder depending on whether its a volume, snapshot or template. Here i'm skipping the copy from swift if the file is already there.